pkg/deploy/registry/deployconfig/etcd/etcd.go

```diff
@@ -152,48 +127,17 @@ func (r *ScaleREST) Update(ctx kapi.Context, obj runtime.Object) (runtime.Object
 		return nil, false, errors.NewNotFound("scale", scale.Name)
 	}
 
-	scaleRet := &extensions.Scale{
-		ObjectMeta: kapi.ObjectMeta{
-			Name:              deploymentConfig.Name,
-			Namespace:         deploymentConfig.Namespace,
-			CreationTimestamp: deploymentConfig.CreationTimestamp,
-		},
-		Spec: extensions.ScaleSpec{
-			Replicas: scale.Spec.Replicas,
-		},
-		Status: extensions.ScaleStatus{
-			Selector: deploymentConfig.Spec.Selector,
-		},
-	}
-
-	// TODO(directxman12): this is going to be a bit out of sync, since we are calculating it
-	// here and not as part of the deploymentconfig loop -- is there a better way of doing it?
-	totalReplicas, err := r.replicasForDeploymentConfig(deploymentConfig.Namespace, deploymentConfig.Name)
-	if err != nil {
-		return nil, false, err
-	}
+	scaleRet := api.ScaleFromConfig(deploymentConfig)
 
+	totalReplicas := deploymentConfig.Status.Replicas
 	oldReplicas := deploymentConfig.Spec.Replicas
 	deploymentConfig.Spec.Replicas = scale.Spec.Replicas
 	if err := r.registry.UpdateDeploymentConfig(ctx, deploymentConfig); err != nil {
 		return nil, false, err
 	}
+
+	scaleRet.Spec.Replicas = scale.Spec.Replicas
 	scaleRet.Status.Replicas = totalReplicas + (scale.Spec.Replicas - oldReplicas)
```

Comment:
With @ironcladlou's change to how DeploymentConfig scaling works, I don't think this calculation is accurate any more. We should just keep `TotalReplicas == oldReplicas`, since the replica count on RCs controlled by the DC won't change until the DC controller updates them.